shakenfist/virt.py

```diff
@@ -68,106 +53,280 @@ def _safe_int_cast(i):
 
 
 class Instance(object):
-    def __init__(self, db_entry):
-        self.db_entry = db_entry
+    def __init__(self, cpus, memory, name, namespace, uuid,
+                 block_devices=None, devices=None, console_port=0,
+                 disk_spec=None, error_message=None, node=None,
+                 node_history=None, placement_attempts=0, power_state='',
+                 power_state_previous=None, power_state_updated=0,
+                 requested_placement=None, ssh_key=None, state=None,
+                 state_updated=None, user_data=None, vdi_port=0,
+                 video=None):
+
+        self.uuid = uuid
+        self.memory = memory
+        self.name = name
+        self.namespace = namespace
+        self.block_devices = block_devices
+        self.devices = devices
+        self.console_port = console_port
+        self.cpus = cpus
+        self.disk_spec = disk_spec
+        self.error_message = error_message
+        self.node = node
+        self.node_history = node_history
+        self.placement_attempts = placement_attempts
+        self.power_state = power_state
+        self.power_state_previous = power_state_previous
+        self.power_state_updated = power_state_updated
+        self.requested_placement = requested_placement
+        self.ssh_key = ssh_key
+        self.state = state
+        self.state_updated = state_updated
+        self.user_data = user_data
+        self.vdi_port = vdi_port
+        self.video = video
```

Comment:
For these, what would you think about using properties a bit more to hide the setter methods?

```
$ cat accessor.py 
#!/usr/bin/python3

class C(object):
    def __init__(self, x):
        self.__set_x(x)

    def __get_x(self):
        return self._x

    def __set_x(self, x):
        self._x = x

    x = property(__get_x, __set_x)


foo = C(1)
print(foo.x)
foo.x = 2
print(foo.x)
```

Which would return:

```
$ python3 accessor.py 
1
2
```

Notably this would not work with the setters where we take more than one argument, but I am unsure how common that is without an audit.

Reply:
I like that idea. Will make it so!